src/librustc/ty/mod.rs

```diff
@@ -2746,12 +2746,19 @@ fn associated_item_def_ids<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     Rc::new(vec)
 }
 
+fn describe_def<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
+                          def_id: DefId)
+                          -> Option<Def> {
+    ty::queries::describe_def::get(tcx, DUMMY_SP, def_id)
```

Comment:
This is just an infinite loop though. You're providing the `describe_def` query for the local crate as querying `describe_def` again with the same key.